lib/Parse/ParseExpr.cpp

```diff
@@ -79,6 +79,16 @@ ParserResult<Expr> Parser::parseExprIs() {
   return makeParserResult(IsExpr::create(Context, isLoc, type.get()));
 }
 
+/// parseExprIsPattern
+ParserResult<Expr> Parser::parseExprIsCase() {
+  SourceLoc isLoc = consumeToken(tok::kw_is);
+  SourceLoc caseLoc = consumeToken(tok::kw_case);
+  
+  ParserResult<Pattern> patternResult = parseMatchingPattern(true, /*inIsCaseExpr*/true);
```

Comment:
Does this handle patterns like `_` or identifiers?

I'll run some tests later, but from just running a mental interpreter I think we might want to use `parsePattern()` instead.

Reply:
This should be fine, it'll be resolved by pattern resolution. AFAIK `parsePattern` is only used for binding patterns, e.g `let <pattern> = <expr>`. So `parseMatchingPattern` is correct here.